atlasdb-tests-shared/src/main/java/com/palantir/test/utils/SubdirectoryCreator.java

```diff
@@ -14,16 +14,22 @@
  * limitations under the License.
  */
 
-package com.palantir;
+package com.palantir.test.utils;
 
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
 import java.io.File;
 
-public class SubdirectoryCreator {
+public final class SubdirectoryCreator {
+
+    private SubdirectoryCreator() {
+        throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
```

Comment:
I don't think we need to throw the exception here. In general we're ok with utility classes with an empty body in the private constructor.